tools/orca/src/models/read_write_attr.rs

```diff
@@ -0,0 +1,104 @@
+use crate::model::{self, ActorModel, Transition, TransitionAction, TransitionResult};
+
+use crate::error::Error;
+use crate::run::EventRecord;
+use crate::state::*;
+use kanidm_client::KanidmClient;
+
+use async_trait::async_trait;
+
+use std::time::Duration;
+
+enum State {
+    Unauthenticated,
+    Authenticated,
+    ReadAttribute,
+    WroteAttribute,
+}
+
+pub struct ActorReadWrite {
+    state: State,
+}
+
+impl ActorReadWrite {
+    pub fn new() -> Self {
+        ActorReadWrite {
+            state: State::Unauthenticated,
+        }
+    }
+}
+
+#[async_trait]
+impl ActorModel for ActorReadWrite {
+    async fn transition(
+        &mut self,
+        client: &KanidmClient,
+        person: &Person,
+    ) -> Result<EventRecord, Error> {
+        let transition = self.next_transition();
+
+        if let Some(delay) = transition.delay {
+            tokio::time::sleep(delay).await;
+        }
+
+        // Once we get to here, we want the transition to go ahead.
+        let (result, event) = match transition.action {
+            TransitionAction::Login => model::login(client, person).await,
+            TransitionAction::ReadAttribute => model::person_get(client, person).await,
+            TransitionAction::WriteAttribute => model::person_set(client, person).await,
+            TransitionAction::Logout => model::logout(client, person).await,
+            _ => Err(Error::InvalidState),
+        }?;
+
+        // Given the result, make a choice about what text.
```

Comment:
About what text?

Reply:
Just some absent minded cut & paste from the Basic model :p